haskell/src/Monocle/Client.hs

```diff
@@ -67,11 +69,13 @@ withClient ::
   Text ->
   -- | An optional manager
   Maybe Manager ->
+  -- | An option Auth token
+  Maybe Text ->
   -- | The callback
   (MonocleClient -> m a) ->
   -- | withClient performs the IO
   m a
-withClient url managerM callBack =
+withClient url managerM tokenM callBack =
```

Comment:
Should we read the token from the environment instead? Otherwise `withClient` user will all have to do that instead.